### PR TITLE
fix: do not delete payments

### PR DESCRIPTION
**Pull request by dni**
as a safety measure we do not delete failed payments

### CODE REVIEW

**Comment by motorina0** [comment]
Great PR. This was a performance improvement with serious negative implications.
If an outgoing payment is deleted by mistake then the balance of the LNbits server would be higher than the one on the Funding Source.

The deletion of old not successful payments should happen outside the main business logic of the application.
There is a command for deleted old (more than 6 months) empty wallets. 
There should be a command to delete old non-successful payments. @dni can you create one (I believe you also handled the wallets command).

**Review by motorina0 — APPROVED** [review]

**Review by talvasconcelos — APPROVED** [review]

**Review by arcbtc — APPROVED** [review]

**Review by arcbtc — CHANGES_REQUESTED** [review]
This is a great idea. I was worried about reintroducing the infinite pending transaction issue, but thats something to tackle after we merge this important update. 
Sorry for not appreciating the importance of this pr originally 🙏

File: lnbits/core/models.py [hunk 1]
```diff
@@ -253,58 +253,25 @@ async def set_pending(self, pending: bool) -> None:
...
+
         funding_source = get_funding_source()
         if self.is_out:
             status = await funding_source.get_payment_status(self.checking_id)
         else:
             status = await funding_source.get_invoice_status(self.checking_id)
 
         logger.debug(f"Status: {status}")
-
-        if self.is_in and status.pending and self.is_expired and self.expiry:
```
Comment (motorina0) [inline lnbits/core/models.py:276]: good clean-up?

Comment (dni) [reply lnbits/core/models.py:276]: yes is see no reason to keep this in the model class

**Pull request closed without merging**